src/render_engine/vulkan/vulkan_render_engine.cpp

```diff
@@ -1391,17 +1443,30 @@ namespace nova::renderer::rhi {
         vkGetDeviceQueue(device, copy_family_idx, 0, &copy_queue);
     }
 
-    bool VulkanRenderEngine::does_device_support_extensions(VkPhysicalDevice device) {
+    bool VulkanRenderEngine::does_device_support_extensions(VkPhysicalDevice device, const std::vector<char*>& required_device_extensions) {
         uint32_t extension_count;
         vkEnumerateDeviceExtensionProperties(device, nullptr, &extension_count, nullptr);
         std::vector<VkExtensionProperties> available(extension_count);
         vkEnumerateDeviceExtensionProperties(device, nullptr, &extension_count, available.data());
 
-        std::set<std::string> required = {VK_KHR_SWAPCHAIN_EXTENSION_NAME};
+        std::set<std::string> required;
+        for(const auto* extension : required_device_extensions) {
+            required.emplace(extension);
+        }
+
         for(const auto& extension : available) {
             required.erase(static_cast<const char*>(extension.extensionName));
         }
 
+        if(!required.empty()) {
+            std::stringstream ss;
+            for(const auto& extension : required) {
+                ss << extension;
```

Comment:
Do the extension names have spaces after them, if not, you're just going to get a lump of text.

Reply:
Fixed